packages/lit-dev-content/site/docs/components/lifecycle.md

```diff
@@ -357,22 +357,79 @@ window.onunhandledrejection = function(e) {
+#### scheduleUpdate() {#scheduleupdate}
 
-Call `performUpdate()` to immediately process a pending update. This should generally not be needed, but it can be done in rare cases when you need to update synchronously.
+Override `scheduleUpdate()` to customize the timing of the update cycle. `scheduleUpdate()` is called when you request an update, and by default it calls `performUpdate()` immediately. Override it to defer the update. For example, you could use it to coordinate update timing between multiple elements.
+
+The following code schedules the update to occur just before the next frame:
+
+{% switchable-sample %}
+
+```ts
+protected override async scheduleUpdate(): Promise<void> {
+  await new Promise<void>((resolve) => 
+      requestAnimationFrame(() => resolve()));
+  this.performUpdate();
+}
+```
+
+```js
+async scheduleUpdate() {
+  await new Promise((resolve) => requestAnimationFrame(() => resolve()));
+  this.performUpdate();
+}
+```
+
+{% endswitchable-sample %}
+
+If you override `scheduleUpdate()`, it's your responsibility to call `performUpdate()`.
+
+In an async function, you can `await` any promises, then call `performUpdate()` to complete the current update. The *next* update can't proceed until `scheduleUpdate()` returns.
 
-Implement `performUpdate()` to customize the timing of the update cycle. This can be useful for implementing custom scheduling. Note, if `performUpdate()` returns a Promise, the `updateComplete` Promise will await it.
+You can also write `scheduleUpdate()` as a regular function. In this case, `scheduleUpdate()` should return a promise. When resolving the promise, you must call `performUpdate()` to complete the current update. The *next* update can't proceed until the promise resolves.
+
+{% switchable-sample %}
+
+```ts
+protected override scheduleUpdate(): Promise<void> {
+  return new Promise<void>((resolve) => 
+      requestAnimationFrame(() => {
+        this.performUpdate();
+        resolve()
+      ));
+}
+```
 
 ```js
-async performUpdate() {
-   await new Promise((resolve) => setTimeout(() => resolve()));
-   return super.performUpdate();
+scheduleUpdate():  {
+  return new Promise((resolve) => 
+      requestAnimationFrame(() => {
+        this.performUpdate();
+        resolve()
+      )
+  );
 }
 ```
 
-In this example, the update is performed after paint. This technique can be used to unblock the main rendering/event thread. See the Chrome Dev Summit talk by Justin Fagnani [The Virtue of Laziness](https://www.youtube.com/watch?v=ypPRdtjGooc) for an extended discussion.
+{% endswitchable-sample %}
+
+This technique can be used to unblock the main rendering/event thread. See the Chrome Dev Summit talk by Justin Fagnani [The Virtue of Laziness](https://www.youtube.com/watch?v=ypPRdtjGooc) for an extended discussion. This talk shows an early version of Lit, but describes many useful concepts.
```

Comment:
I feel like this maybe could be merged with the intro above to shorten the whole section.

Reply:
I played around with this, but putting it up top kind of suggests that you need to go watch the video to understand the section. The video has some good information, but it's a little confusing as an intro because it covers several topics (including shceduleUpdate) and it uses older versions of lit-html and lit-element. (I think it also mentions an add-on library.)

Reply:
Got it. I can do that.